includes/class-wc-payments.php

```diff
@@ -159,7 +159,16 @@ public static function init() {
 
 		if ( ! self::check_plugin_dependencies( true ) ) {
 			add_filter( 'admin_notices', [ __CLASS__, 'check_plugin_dependencies' ] );
-			return;
+
+			/**
+			 * After displaying notice errors in WP Admin, still silently load the plugin if the account is connected.
+			 *
+			 * @since 3.1.0
+			 */
+			$account_data = get_option( 'wcpay_account_data' );
+			if ( ! isset( $account_data['account'] ) ) {
+				return;
+			};
```

Comment:
Yeah, ideally we still use `self::$api_client->is_stripe_connected()`. 

> I wonder, could we move the initialization of `self::$api_client` and `self::$account` before the call to `check_plugin_dependencies` [...]

I will explore this path and see if that's possible.

Reply:
> I wonder, could we move the initialization of self::$api_client and self::$account before the call to check_plugin_dependencies [...]

The flow of `self::$api_client->is_stripe_connected()` looks like the following: 

- WC_Payments_Account->is_stripe_connected()
- WC_Payments_Account->try_is_stripe_connected()
- WC_Payments_Account->get_cached_account_data()
- WC_Payments_API_Client->get_account_data()
- WC_Payments::get_gateway()->is_in_dev_mode()

For the last call above, it's tricky as it requires WC_Payments::$card_gateway assigned value, which is only available very late:

https://github.com/Automattic/woocommerce-payments/blob/2d76e585e837cdeabd5f5a91f65f0dd565a0d47e/includes/class-wc-payments.php#L267-L269 

If we move this gateway code before the call to `check_plugin_dependencies`, it will create a fatal error for the case WooCommerce does not exist.

Reply:
I've checked my current code, and found out that it will yell fatal errors in this case:

- WooCommerce is not active.
- WCPay cache account exists.

I tried to change this behavior around [the check for old WC versions](https://github.com/Automattic/woocommerce-payments/blob/2d76e585e837cdeabd5f5a91f65f0dd565a0d47e/includes/class-wc-payments.php#L472-L494) but could not find a proper way due to the dual purposes of the current `check_plugin_dependencies` code. May you have some other ideas? 

That does not mean this change is not possible but it requires more thought and careful testing.

**All in all, we may not change anything here now as it might create more issues with this change.**

Reply:
> In general, it feels like the test/dev mode checks shouldn't belong in the payment gateway settings anymore because our plugin is no longer just a payment gateway.

Agreed here! 

> I also tested and it looks like we manage to populate the account before deactivating the plugin, so maybe auto-update will have the same effect: it would populate the cache and then update, so things can keep working.

I will double-check this.

> Edit: but that would not render the notices for people who are connected but out of date 🤔 Maybe it is time to split the `check_plugin_dependencies` into two - one for returning true/false and one for rendering the notice.

What's about the following idea? 

- `check_plugin_dependencies()` stops using param `$silent`, and returns an array including two keys: `passed` with a boolean value and `message`. 
- When calling `check_plugin_dependencies()`, we display an admin error notice for `message`, and continue loading the whole plugin or not based on `passed`.

Reply:
@marcinbot. I've implemented my idea and your suggestion:

- a453a88e09c39331642397de848dd3e73ba3fcb3 - `check_plugin_dependencies` returns `message` and `passed` flag. 
- f03f18c50d1a18b909247595d6f96b3bc7d4bf56 - use `has_cached_account_connection` as your suggestion.